File: apigateway_getting_started/create_api_client_resource_child_get_test.js
import { APIGatewayClient, TestInvokeMethodCommand } from "@aws-sdk/client-api-gateway";
(async function () {
    const apig = new APIGatewayClient({ region: "us-east-1" });
    const command = new TestInvokeMethodCommand({
        restApiId: 'abcd1234',
        resourceId: 'bbb222',
        httpMethod: 'GET',
        pathWithQueryString: '/pets/3',
    });
    try {
        const results = await apig.send(command)
        console.log(results)
    } catch (err) {
        console.log("The test on 'GET /pets/{petId}' method failed:\n", err)
    }
})();

File: apigateway_getting_started/create_api_client_resource_child_get_test.py
import botocore
import boto3
import logging

logger = logging.getLogger()
apig = boto3.client("apigateway")

try:
    result = apig.test_invoke_method(
        restApiId="abcd1234",
        resourceId="bbb222",
        httpMethod="GET",
        pathWithQueryString="/pets/3",
    )
except botocore.exceptions.ClientError as error:
    logger.exception("Test invoke method on 'GET /pets/{petId}' failed: %s", error)
    raise
print(result)
